### DISCUSSION TITLE
Cache based on external dir? (but don't use globalDeps)

### DISCUSSION THREAD

**Discussion by zachrip** [vercel/turborepo#929] [Help] [answered]
Hello, I have a project which is laid out like this:
```
root
    proto
    ts
        apps
            foo
            bar
            ui-app (does *not* rely on proto and is slow to build)
        packages
            ts-proto
```
The apps rely on the ts-proto package and ts-proto relies on proto dir. However, since proto is not in the same dir as ts-proto, when changes are made to it, turbo thinks nothing has changed. Adding `"globalDependencies": ["proto/**/*.proto"]` fixes it, but it causes every package to rebuild which is not what I want as only a subset of the projects rely on the proto.

**Comment by anthonyshew** [comment, marked as answer]
My first suggestion would be to move the `.proto` output into your workspace if you can. This can help with ensuring your packages are self-contained as needed.

If you can't do that, I'd say use the `inputs` key and point it to your output folder (that is outside of your workspaces). Also, make sure that the outputs you're creating are indeed in the `outputs` key of your task that produces it.
